### ISSUE TITLE
`initialize` stuck indefinitely on second call after upgrading to 1.1.1

### ISSUE THREAD

**Issue by beetlebum** [DataDog/dd-sdk-reactnative#282]
We've just tried upgrading from 1.0.0 to 1.1.1.
One use case seems to not work anymore:
- When our application starts, we call DataDog `initialize`
- When a user logs out, we restart React Native, wiping all local state. After restarting, DataDog `initialize` will be called a second time and after the upgrade, it gets stuck in an infinite loop.

```
System:
    OS: macOS 12.6
    CPU: (8) arm64 Apple M1 Pro
    Memory: 119.00 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.3.1 - ~/.nvm/versions/node/v16.14.0/bin/npm
    Watchman: 2022.08.15.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 16.0, macOS 12.3, tvOS 16.0, watchOS 9.0
    Android SDK:
      API Levels: 27, 28, 29, 31, 32, 33
      Build Tools: 30.0.2, 30.0.3, 31.0.0, 32.0.0, 32.0.0, 33.0.0, 33.0.0
      System Images: android-33 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: Dolphin 2021.3.1 Dolphin 2021.3.1
    Xcode: 14.0/14A309 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.13 - /Users/diegolopez/.sdkman/candidates/java/current/bin/javac
  npmPackages:
    @react-native-community/cli: ^9.0.0 => 9.1.1
    react: ^18.2.0 => 18.2.0
    react-native: ^0.70.0 => 0.70.1
```

**Comment by louiszawadzki** [comment]
Hi @beetlebum, thanks for reaching out!

Unfortunately we could not reproduce this bug, could you share either more information to help us help you on this issue?

- What part of the code is actually stuck in an infinite loop?
- What config parameters to you pass to the `initialize` method?
- What methods do you call to restart React Native?

Alternatively, you can also share a minimal app reproducing the issue.


Thanks!

**Comment by beetlebum** [comment]
<img width="560" alt="image" src="https://user-images.githubusercontent.com/227258/191714998-7db2254f-d5c9-4eff-b592-f71ff9999499.png">

You can see the initialisation code above.

When logging out, we usually clear accounts from third party SDKs, but I believe DataDog does not provide a way to do that.
Then the application starts again, first Sentry is initialised, then DataDog, then CodePush and then the rest of the app.

The issue only started happening with 1.1.1, reverting to 1.0.0 has fixed it

**Comment by louiszawadzki** [comment]
Hi @beetlebum,

I think I might have an idea, but I don't know for sure if it will fix your issue, could you try it and tell me if it works?

Open your `.xcworkspace` in XCode, then edit the DdSdk.swift file in Pods > Development Pods > DatadogSDKReactNative:
<img width="1252" alt="image" src="https://user-images.githubusercontent.com/8973379/191734170-666122ab-514f-4b99-8554-18e0e406a23b.png">

In the `initialize` method, replace the `return` by `resolve(nil)`.

If it works for you, I will open a PR for a fix and release it under 1.1.2 :) 

Thanks!
